pinot-common/src/main/java/org/apache/pinot/common/utils/ServiceStatus.java

```diff
@@ -243,11 +278,18 @@ public synchronized Status getServiceStatus() {
         }
 
         // Resource is done starting up, remove it from the set
-        iterator.remove();
```

Comment:
There is a return in line 275 - should that be a continue (as we want to track the % of resources online and not get stuck on just one, right?)

Reply:
Nope. the return should be as it is, because we want to stop when we reach a bad resource. We already check the threshold at the top of the loop, so if we are inside the loop the threshold is not met yet. We check only one resource each time in the loop